src/Monitor/Monitor/ScheduledQueryRules/UpdateScheduledQueryRuleCommand.cs

```diff
@@ -0,0 +1,138 @@
+using Microsoft.Azure.Management.Internal.Resources.Utilities.Models;
+using Microsoft.Azure.Management.Monitor;
+using Microsoft.WindowsAzure.Commands.Utilities.Common;
+using Microsoft.Azure.Management.Monitor.Management.Models;
+using Microsoft.WindowsAzure.Commands.Common.CustomAttributes;
+
+namespace Microsoft.Azure.Commands.Insights.ScheduledQueryRules
+{
+    /// <summary>
+    /// Updates a ScheduledQueryRule object
+    /// </summary>
+    [Cmdlet(VerbsData.Update, ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "ScheduledQueryRule",
+         SupportsShouldProcess = true, DefaultParameterSetName = ByRuleName), OutputType(typeof(PSScheduledQueryRuleResource))]
+    public class UpdateScheduledQueryRuleCommand : ManagementCmdletBase
+    {
+
+        private const string ByInputObject = "ByInputObject";
+        private const string ByRuleName = "ByRuleName";
+        private const string ByResourceId = "ByResourceId";
+
+        #region Cmdlet parameters
+
+        [Parameter(ParameterSetName = ByInputObject, Mandatory = true, ValueFromPipeline = true,
+            HelpMessage = "The Scheduled Query Rule resource")]
+        [ValidateNotNull]
+        public PSScheduledQueryRuleResource InputObject { get; set; }
+
+        [Parameter(ParameterSetName = ByResourceId, Mandatory = true, ValueFromPipelineByPropertyName = true,
+            HelpMessage = "The resource Id")]
+        [ValidateNotNullOrEmpty]
+        public string ResourceId { get; set; }
+
+        /// <summary>
+        /// Alert name
+        /// </summary>
+        [Parameter(ParameterSetName = ByRuleName, Mandatory = true, ValueFromPipelineByPropertyName = true,
+            HelpMessage = "The alert name")]
+        [ResourceNameCompleter("Microsoft.insights/scheduledqueryrules", nameof(ResourceGroupName))]
+        [ValidateNotNullOrEmpty]
+        public string RuleName { get; set; }
+
+        /// <summary>
+        /// The resource group name
+        /// </summary>
+        [Parameter(ParameterSetName = ByRuleName, Mandatory = true, ValueFromPipelineByPropertyName = true,
+            HelpMessage = "The resource group name")]
+        [ResourceGroupCompleter]
+        [ValidateNotNullOrEmpty]
+        public string ResourceGroupName { get; set; }
+
+        /// <summary>
+        /// Alert status - enabled or not, supported values - "true", "false"
+        /// </summary>
+        [Parameter(ParameterSetName = ByRuleName, Mandatory = true,
+            HelpMessage = "The azure alert state - valid values - true, false")]
+        [Parameter(ParameterSetName = ByInputObject, Mandatory = true,
+            HelpMessage = "The azure alert state - valid values - true, false")]
+        [Parameter(ParameterSetName = ByResourceId, Mandatory = true,
+            HelpMessage = "The azure alert state - valid values - true, false")]
+        [ValidateSet("true", "false")]
+        [PSArgumentCompleter("true", "false")]
+        public string Enabled { get; set; }
```

Comment:
@shpimpal same comment about changing this to `SwitchParameter`

Reply:
@shpimpal same comment here about changing this to a `bool` and removing the attributes